packages/cli/src/commands/addPlatform/addPlatform.ts

```diff
@@ -0,0 +1,259 @@
+import {
+  CLIError,
+  getLoader,
+  logger,
+  prompt,
+} from '@react-native-community/cli-tools';
+import {Config} from '@react-native-community/cli-types';
+import {join} from 'path';
+import {readFileSync} from 'fs';
+import chalk from 'chalk';
+import {install, PackageManager} from './../../tools/packageManager';
+import npmFetch from 'npm-registry-fetch';
+import semver from 'semver';
+import {checkGitInstallation, isGitTreeDirty} from '../init/git';
+import {changePlaceholderInTemplate} from '../init/editTemplate';
+import {
+  copyTemplate,
+  executePostInitScript,
+  getTemplateConfig,
+  installTemplatePackage,
+} from '../init/template';
+import {tmpdir} from 'os';
+import {mkdtempSync} from 'graceful-fs';
+import {existsSync} from 'fs';
+
+type Options = {
+  packageName: string;
+  version: string;
+  pm: PackageManager;
+  title: string;
+};
+
+const NPM_REGISTRY_URL = 'http://registry.npmjs.org'; // TODO: Support local registry
+
+const getAppName = async (root: string) => {
+  logger.log(`Reading ${chalk.cyan('name')} from package.json…`);
+  const pkgJsonPath = join(root, 'package.json');
+
+  if (!pkgJsonPath) {
+    throw new CLIError(`Unable to find package.json inside ${root}`);
+  }
+
+  let name;
+
+  try {
+    name = JSON.parse(readFileSync(pkgJsonPath, 'utf8')).name;
+  } catch (e) {
+    throw new CLIError(`Failed to read ${pkgJsonPath} file.`, e as Error);
+  }
+
+  if (!name) {
+    const appJson = join(root, 'app.json');
+    if (appJson) {
+      logger.log(`Reading ${chalk.cyan('name')} from app.json…`);
+      try {
+        name = JSON.parse(readFileSync(appJson, 'utf8')).name;
+      } catch (e) {
+        throw new CLIError(`Failed to read ${pkgJsonPath} file.`, e as Error);
+      }
+    }
+
+    if (!name) {
+      throw new CLIError('Please specify name in package.json or app.json.');
+    }
+  }
+
+  return name;
+};
+
+const getPackageMatchingVersion = async (
+  packageName: string,
+  version: string,
+) => {
+  const npmResponse = await npmFetch.json(packageName, {
+    registry: NPM_REGISTRY_URL,
```

Comment:
registry should be configurable or read from npmrc or equivalent

Reply:
agree, it turns out that we have a util for that 👍